### ISSUE TITLE
Desktop App Beta Testing

### ISSUE THREAD

**Issue by kspearrin** [bitwarden/desktop#13]
We are releasing the Bitwarden desktop application as a public beta to gather feedback and fix any issues before going live. The application is now available on Windows, macOS, and Linux distributions.

![My Vault](http://imgur.com/fdhNCJl.png "My Vault")

# Download

You can download the desktop app from the GitHub releases page here: 

### https://github.com/bitwarden/desktop/releases

Choose the appropriate file with the most recent version for your operating system:

- **Windows**: `Bitwarden-Installer-x.y.z.exe`
- **macOS**: `Bitwarden-x.y.z.dmg`
- **Linux**: `Bitwarden-x.y.z-x86_64.AppImage`
  - Note: Make sure you adjust permissions to make the AppImage executable: `chmod a+x Bitwarden*.AppImage`. More info at https://github.com/AppImage/AppImageKit/wiki

Other available installations:

- **Windows Portable**: `Bitwarden-Portable-x.y.z.exe`
  - Note: The portable version does not auto-update. Manual updates are required  (re-downloading a new `.exe`).
- **Windows Chocolatey**: *coming soon*
- **macOS homebrew cask**: `brew cask install bitwarden`
- **macOS Mac App Store**: *coming soon*
- **Linux .deb**: `Bitwarden-x.y.z-amd64.deb`
  - Note: `.deb` installations do not auto-update. It is recommended to use the `.AppImage` version so that you do not have to update the application manually.
- **Linux .rpm**: `Bitwarden-x.y.z-x86_64.rpm`
  - Note: `.rpm` installations do not auto-update. It is recommended to use the `.AppImage` version so that you do not have to update the application manually.
- **Linux snap**: *coming soon*

# Automatic Updates

Unless otherwise noted above, the app will automatically alert you about updates and install them as they become available.

# Report Problems / Feedback

Please open a new issue in this repository if you find any problems.

https://github.com/bitwarden/desktop/issues

# Known Issues

- ~~The Windows `exe` files are not currently signed. Because of this, you may get a warning when trying to open them for the first time. Select the "More Info" link to "Run Anyways". The files will be properly signed when we go live.~~
- The application menu is permanently disabled on Ubuntu 16.04. ref: https://github.com/electron/electron/issues/11891

**Comment by Vistaus** [comment]
Awesome work, thank you so much! :) I'll wait for the Snap to arrive though.

**Comment by Attoy** [comment]
Thanks @kspearrin , I'll download it and start testing it right away!

**Comment by peterkeanu** [comment]
Settings don't work in Mac...

**Comment by papertigers** [comment]
Any chance you will officially support FreeBSD?

**Comment by kspearrin** [comment]
@papertigers I don't really know anything about FreeBSD. What is needed to support it? Will the AppImage not work there?

**Comment by papertigers** [comment]
@kspearrin you can probably ignore my request for now.  It looks like electron currently doesn't build on FreeBSD.  There are some community patches that need to land first.

**Comment by kspearrin** [comment]
@papertigers I ran our packaging tool with the `freebsd` option enabled and it produced this. Does it work?

https://github.com/bitwarden/desktop/releases/download/v0.0.7/bitwarden-0.0.7.freebsd

**Comment by papertigers** [comment]
@kspearrin it looks like its an xz file and if I untar everything, then `file` says its `Bitwarden/bitwarden: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, stripped`

**Comment by EddieFAF** [comment]
What's the difference between bitwarden-x.y.z.exe and bitwarden-x.y.z-x64.nsis.7z?

**Comment by kspearrin** [comment]
@EddieFAF One is a packaged EXE and the other is not.

**Comment by kspearrin** [comment]
@papertigers So it doesn't work?

**Comment by zhtw2013** [comment]
Menu bar language translation?
![001](https://user-images.githubusercontent.com/9501406/36482488-6e65e326-174e-11e8-8493-848190766021.png)

**Comment by kspearrin** [comment]
@zhtw2013 Fixed in next version.

**Comment by rtloeffler** [comment]
@kspearrin thanks much man, a nice surprise! 

One thing I like on the web version is when viewing "ALL" it'll still split up each folder. Maybe something we can see on the desktop app.

**Comment by JSN190** [comment]
It's looking nice so far. Maybe we could have a flatpak package too.

**Comment by kspearrin** [comment]
@zhtw2013 Can you confirm the menu is translated now in 0.0.8?

**Comment by zhtw2013** [comment]
![image](https://user-images.githubusercontent.com/9501406/36568266-4e644ab0-1864-11e8-8e83-366044f3a898.png)

![image](https://user-images.githubusercontent.com/9501406/36568317-7c6a12a0-1864-11e8-9c26-84a79b6f87f1.png)

![image](https://user-images.githubusercontent.com/9501406/36568338-8ac62672-1864-11e8-8f88-f91eb7df57dd.png)

Menu Is translated!

But
Wndows、Help、Minimize、Close、Zoom in、Zoom out...
Can you add string to crowdin?

**Comment by dabura667** [comment]
I can confirm some strings are missing from Crowdin.

Japanese is also partially English currently.

**Comment by kspearrin** [comment]
Ok, what strings ?

**Comment by dabura667** [comment]
Quite a bit.

Lemme take a ton of screenshots :-P

**Comment by dabura667** [comment]
![_001](https://user-images.githubusercontent.com/6586275/36637065-59f029ee-1a17-11e8-9475-894271b1c096.png)
![_002](https://user-images.githubusercontent.com/6586275/36637078-a4a32c52-1a17-11e8-8813-9a1cd05db339.png)
![_003](https://user-images.githubusercontent.com/6586275/36637080-a7f4d8e2-1a17-11e8-890f-21cc420f2be3.png)
![_004](https://user-images.githubusercontent.com/6586275/36637081-ab774d56-1a17-11e8-93f9-307778c81dcf.png)
![_005](https://user-images.githubusercontent.com/6586275/36637083-ae7e4c66-1a17-11e8-8fe0-c6629481403e.png)
![_006](https://user-images.githubusercontent.com/6586275/36637085-b157dac4-1a17-11e8-9391-1c581f805439.png)
![_007](https://user-images.githubusercontent.com/6586275/36637088-b8e2b02a-1a17-11e8-9095-0ec2bf7144f3.png)
![_008](https://user-images.githubusercontent.com/6586275/36637090-bdee2c5c-1a17-11e8-90f4-69e8c4f5b00d.png)
![_009](https://user-images.githubusercontent.com/6586275/36637091-c22bad3a-1a17-11e8-9510-8ee2eba371b9.png)
![_010](https://user-images.githubusercontent.com/6586275/36637092-c567f198-1a17-11e8-81de-1b407b094f00.png)
![_011](https://user-images.githubusercontent.com/6586275/36637093-c81142e6-1a17-11e8-9e10-5f1775f36363.png)
![_012](https://user-images.githubusercontent.com/6586275/36637094-ca204578-1a17-11e8-9939-e39092111c47.png)
![_013](https://user-images.githubusercontent.com/6586275/36637095-cc1d3f2a-1a17-11e8-9734-bb7a257bbf92.png)

**Comment by kspearrin** [comment]
These strings are not missing from crowdin. They just haven’t been translated yet.

**Comment by dabura667** [comment]
@kspearrin I just checked desktop on crowdin, and it was 100%???

**Comment by dabura667** [comment]
![_014](https://user-images.githubusercontent.com/6586275/36637118-22c1e77c-1a18-11e8-9ed9-172dcb9a4c2d.png)
Also, Linux AppImage doesn't have an icon.

**Comment by kspearrin** [comment]
Then they were likely recently translated. It probably needs to just be refreshed for the next build. It’s not automatic.

**Comment by dabura667** [comment]
Oh yes, I search for some of the words on this and it says "a day ago"

I will wait for the new release. Great job!

Once desktop is settled down will you crowd fund a formal audit? Thanks.

**Comment by dabura667** [comment]
for the browser icon, see this comment on a similar issue for a workaround.

https://github.com/electron-userland/electron-builder/issues/2269#issuecomment-342168989

**Comment by zhtw2013** [comment]
![image](https://user-images.githubusercontent.com/9501406/36639822-6c811770-1a4f-11e8-8ba7-8daf165a66e1.png)

Can you add string to crowdin?
Wndows、Help

**Comment by probonopd** [comment]
I am getting

```
$ Downloads/Bitwarden-0.0.15-x86_64.AppImage 
6211
A JavaScript error occurred in the main process
Uncaught Exception:
Error: ENOENT: no such file or directory, mkdir '/home/me/.config/Bitwarden/logs'
    at Object.fs.mkdirSync (fs.js:891:18)
    at new Main (/tmp/.mount_BitwarWKFVQt/app/resources/app.asar/main.js:304:58)
    at Object.__webpack_require__.constructor.main.main (/tmp/.mount_BitwarWKFVQt/app/resources/app.asar/main.js:339:14)
    at __webpack_require__ (/tmp/.mount_BitwarWKFVQt/app/resources/app.asar/main.js:20:30)
    at /tmp/.mount_BitwarWKFVQt/app/resources/app.asar/main.js:63:18
    at Object.<anonymous> (/tmp/.mount_BitwarWKFVQt/app/resources/app.asar/main.js:66:10)
    at Object.<anonymous> (/tmp/.mount_BitwarWKFVQt/app/resources/app.asar/main.js:1729:3)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
```

on Xubuntu 16.04.

**Comment by kspearrin** [comment]
@probonopd It seems that a change we made in this most recent version is behaving differently on some Linux distros. For some reason the `/home/me/.config/Bitwarden/` folder does not exist yet whenever we try to create the `logs` folder underneath it. We'll get a patch out soon.

**Comment by x80486** [comment]
I'm running Ubuntu (`Linux uplink 4.13.0-36-generic #40~16.04.1-Ubuntu SMP Fri Feb 16 23:25:58 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`). I installed the app via `snap`s, but when I click on the (automatically created) link in the Dash/HUD it doesn't start — but it's launched, so I see that the icon appears in the Launcher but it disappears after some time. I tried several times since I've noticed that, sometimes, there are app(s) that just doesn't run the first couple of times after being installed, e.g.: `KeePassXC`, `Slack`, `DataGrip`...to name a few.

On the other hand, if I go to `/snap/bitwarden/current/` and execute `bitwarden`, I can see the Electron app coming up as expected. So it might be the command executed for the icon.

More info:

```
gorre@uplink:~$ snap version
snap    2.31.1
snapd   2.31.1
series  16
ubuntu  16.04
kernel  4.13.0-36-generic

gorre@uplink:~$ snap list
Name                        Version     Rev   Developer       Notes
bitwarden                   0.0.9       3     bitwarden       -
core                        16-2.31.1   4110  canonical       core
hexchat                     2.12.4      17    tingping        -
keepassxc                   2.2.4       30    keepassxreboot  -
slack                       3.0.5       4     slack           classic
```

**Comment by kspearrin** [comment]
@x80486 We are actively trying to make snap work still. We run into the same issue. See a thread we have open on snap forums here: https://forum.snapcraft.io/t/app-crashes-even-in-devmode-how-do-i-debug-it/4178

**Comment by kspearrin** [comment]
Beta is now over. 1.0.0 will go live soon. Thanks all for the feedback.
